add MAZE-EN + DBWS

IMJoyJ opened the pull request:
add MAZE-EN (not tested)

Lyris12 commented on scripts/MAZE-EN/c100206002.lua:
```diff
@@ -0,0 +1,66 @@
+	e1:SetRange(LOCATION_HAND)
+	e1:SetCountLimit(1,100206002)
+	e1:SetCost(c100206002.thcost)
+	e1:SetTarget(c100206002.thtg)
```
redtext: `Parameter 2 should be "Function".`
```suggestion
	e1:SetTarget(c100206002.target)
```

Lyris12 commented on scripts/MAZE-EN/c100206006.lua:
```diff
@@ -0,0 +1,75 @@
+function c100206006.atktg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsControler(1-tp) and chkc:IsLocation(LOCATION_MZONE) and c100206006.atkfilter(chkc) end
+	if chk==0 then return Duel.IsExistingTarget(c100206006.atkfilter,tp,0,LOCATION_MZONE,1,nil)
+		and c:GetFlagEffect(100206006)==0 end
```
redtext: `attempt to index a nil value (global 'c')`
```suggestion
		and e:GetHandler():GetFlagEffect(100206006)==0 end
```

Lyris12 commented on scripts/MAZE-EN/c100206005.lua:
```diff
@@ -0,0 +1,64 @@
+		and (re:GetActivateLocation() & LOCATION_ONFIELD)>0 and Duel.IsChainDisablable(ev)
+end
+function c100206005.distg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100206005)==0 end
```
redtext: `attempt to index a nil value (global 'c')`
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100206005)==0 end
```

Lyris12 commented on scripts/MAZE-EN/c100206004.lua:
```diff
@@ -0,0 +1,64 @@
+	c:RegisterEffect(e2)
+end
+function c100206004.thfilter(c)
+	return (aux.IsCodeListed(25955164) or aux.IsCodeListed(62340868)
+		or aux.IsCodeListed(98434877)) and c:IsAbleToHand()
```
```suggestion
	return (aux.IsCodeListed(c,25955164) or aux.IsCodeListed(c,62340868)
		or aux.IsCodeListed(c,98434877)) and c:IsAbleToHand()
```

Lyris12 commented on scripts/DBWS-JP/c100420017.lua:
```diff
@@ -0,0 +1,110 @@
+	e1:SetRange(LOCATION_HAND)
+	e1:SetCountLimit(1,100420017)
+	e1:SetCondition(c100420017.spcon)
+	e1:SetCost(c100420017.spcost)
```
redtext: `Parameter 2 should be "Function".`
```suggestion
```

Lyris12 commented on scripts/DBWS-JP/c100420020.lua:
```diff
@@ -0,0 +1,115 @@
+	e1:SetRange(LOCATION_HAND)
+	e1:SetCountLimit(1,100420020)
+	e1:SetCondition(c100420020.spcon)
+	e1:SetCost(c100420020.spcost)
```
redtext: `Parameter 2 should be "Function".`
```suggestion
```

Lyris12 commented on scripts/DBWS-JP/c100420021.lua:
```diff
@@ -0,0 +1,117 @@
+	e1:SetCode(EVENT_FREE_CHAIN)
+	e1:SetProperty(EFFECT_FLAG_CARD_TARGET)
+	e1:SetHintTiming(0,TIMING_END_PHASE)
+	e1:SetCost(c100420021.spcost)
```
redtext: `Parameter 2 should be "Function".`
```suggestion
```

Lyris12 commented on scripts/DBWS-JP/c100420016.lua:
```diff
@@ -0,0 +1,109 @@
+end
+function c100420016.thtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100420016.thfilter,tp,LOCATION_DECK,0,1,nil)
+		and c:GetFlagEffect(100420016)==0 end
```
redtext: `attempt to index a nil value (global 'c')`
```suggestion
		and e:GetHandler():GetFlagEffect(100420016)==0 end
```

Lyris12 commented on scripts/DBWS-JP/c100420016.lua:
```diff
@@ -0,0 +1,109 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420016.indestg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420016)==0 end
```
redtext: `attempt to index a nil value (global 'c')`
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420016)==0 end
```

Lyris12 commented on scripts/MAZE-EN/c100206006.lua:
```diff
@@ -0,0 +1,75 @@
+	if chkc then return chkc:IsControler(1-tp) and chkc:IsLocation(LOCATION_MZONE) and c100206006.atkfilter(chkc) end
+	if chk==0 then return Duel.IsExistingTarget(c100206006.atkfilter,tp,0,LOCATION_MZONE,1,nil)
+		and c:GetFlagEffect(100206006)==0 end
+	c:RegisterFlagEffect(100206006,RESET_CHAIN,0,1)
```
```suggestion
	e:GetHandler():RegisterFlagEffect(100206006,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/MAZE-EN/c100206005.lua:
```diff
@@ -0,0 +1,64 @@
+end
+function c100206005.distg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100206005)==0 end
+	c:RegisterFlagEffect(100206005,RESET_CHAIN,0,1)
```
```suggestion
	e:GetHandler():RegisterFlagEffect(100206005,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420016.lua:
```diff
@@ -0,0 +1,109 @@
+function c100420016.thtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100420016.thfilter,tp,LOCATION_DECK,0,1,nil)
+		and c:GetFlagEffect(100420016)==0 end
+	c:RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```
```suggestion
	e:GetHandler():RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420016.lua:
```diff
@@ -0,0 +1,109 @@
+end
+function c100420016.indestg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420016)==0 end
+	c:RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```
```suggestion
	e:GetHandler():RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420016.lua:
```diff
@@ -0,0 +1,109 @@
+end
+function c100420016.destg(e,tp,eg,ep,ev,re,r,rp,chk)
+	local g=e:GetHandler():GetColumnGroup():Filter(Card.IsLocation,nil,LOCATION_MZONE)
+	if chk==0 then return #g>0 and c:GetFlagEffect(100420016)==0 end
+	c:RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return #g>0 and e:GetHandler():GetFlagEffect(100420016)==0 end
	e:GetHandler():RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420017.lua:
```diff
@@ -0,0 +1,110 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420017.indestg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420017)==0 end
+	c:RegisterFlagEffect(100420017,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420017)==0 end
	e:GetHandler():RegisterFlagEffect(100420017,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420017.lua:
```diff
@@ -0,0 +1,110 @@
+end
+function c100420017.destg(e,tp,eg,ep,ev,re,r,rp,chk)
+	local g=e:GetHandler():GetColumnGroup():Filter(Card.IsLocation,nil,LOCATION_SZONE)
+	if chk==0 then return #g>0 and c:GetFlagEffect(100420017)==0 end
+	c:RegisterFlagEffect(100420017,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return #g>0 and e:GetHandler():GetFlagEffect(100420017)==0 end
	e:GetHandler():RegisterFlagEffect(100420017,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420018.lua:
```diff
@@ -0,0 +1,107 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420018.drtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420018)==0
+		and Duel.IsPlayerCanDraw(tp,1) end
+	c:RegisterFlagEffect(100420018,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420018)==0
		and Duel.IsPlayerCanDraw(tp,1) end
	e:GetHandler():RegisterFlagEffect(100420018,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420018.lua:
```diff
@@ -0,0 +1,107 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420018.dmgtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420018)==0 end
+	c:RegisterFlagEffect(100420018,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420018)==0 end
	e:GetHandler():RegisterFlagEffect(100420018,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420019.lua:
```diff
@@ -0,0 +1,120 @@
+end
+function c100420019.thtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100420019.thfilter,tp,LOCATION_DECK,0,1,nil)
+		and c:GetFlagEffect(100420019)==0 end
+	c:RegisterFlagEffect(100420019,RESET_CHAIN,0,1)
```
```suggestion
		and e:GetHandler():GetFlagEffect(100420019)==0 end
	e:GetHandler():RegisterFlagEffect(100420019,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420019.lua:
```diff
@@ -0,0 +1,120 @@
+	return c:IsFaceup() and not c:IsType(TYPE_LINK)
+end
+function c100420019.rthtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420019)==0
+		and Duel.IsExistingMatchingCard(c100420019.rthtgfilter,tp,LOCATION_MZONE,LOCATION_MZONE,1,nil) end
+	c:RegisterFlagEffect(100420019,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420019)==0
		and Duel.IsExistingMatchingCard(c100420019.rthtgfilter,tp,LOCATION_MZONE,LOCATION_MZONE,1,nil) end
	e:GetHandler():RegisterFlagEffect(100420019,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420020.lua:
```diff
@@ -0,0 +1,115 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420020.deftg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420020)==0 end
+	c:RegisterFlagEffect(100420020,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420020)==0 end
	e:GetHandler():RegisterFlagEffect(100420020,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420020.lua:
```diff
@@ -0,0 +1,115 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420020.atktg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420020)==0 end
+	c:RegisterFlagEffect(100420020,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420020)==0 end
	e:GetHandler():RegisterFlagEffect(100420020,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420021.lua:
```diff
@@ -0,0 +1,117 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420021.imtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420021)==0 end
+	c:RegisterFlagEffect(100420021,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420021)==0 end
	e:GetHandler():RegisterFlagEffect(100420021,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420021.lua:
```diff
@@ -0,0 +1,117 @@
+end
+function c100420021.destg(e,tp,eg,ep,ev,re,r,rp,chk)
+	local g=Duel.GetMatchingGroup(aux.TRUE,tp,LOCATION_ONFIELD,LOCATION_ONFIELD,e:GetHandler())
+	if chk==0 then return #g>0 and c:GetFlagEffect(100420021)==0 end
+	c:RegisterFlagEffect(100420021,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return #g>0 and e:GetHandler():GetFlagEffect(100420021)==0 end
	e:GetHandler():RegisterFlagEffect(100420021,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420020.lua:
```diff
@@ -0,0 +1,117 @@
+	e1:SetCode(EFFECT_UPDATE_ATTACK)
+	e1:SetValue(3000)
+	e1:SetReset(RESET_EVENT+RESETS_STANDARD+RESET_PHASE+PHASE_END)
+	tc:RegisterEffect(e1)
```
redtext: `attempt to index a nil value (global 'tc')`

Lyris12 commented on scripts/DBWS-JP/c100420020.lua:
```diff
@@ -0,0 +1,117 @@
+	e1:SetCode(EFFECT_UPDATE_DEFENSE)
+	e1:SetValue(3000)
+	e1:SetReset(RESET_EVENT+RESETS_STANDARD+RESET_PHASE+PHASE_END)
+	tc:RegisterEffect(e1)
```
redtext: `attempt to index a nil value (global 'tc')`

Lyris12 commented on scripts/MAZE-EN/c100206004.lua:
```diff
@@ -0,0 +1,64 @@
+	c:RegisterEffect(e2)
+end
+function c100206004.thfilter(c)
+	return (aux.IsCodeListed(25955164) or aux.IsCodeListed(62340868)
+		or aux.IsCodeListed(98434877)) and c:IsAbleToHand()
```
```suggestion
	return (aux.IsCodeListed(c,25955164) or aux.IsCodeListed(c,62340868)
		or aux.IsCodeListed(c,98434877)) and c:IsAbleToHand()
```

IMJoyJ replied:
This one is weird... I prefer to use "or" because yugioh-wiki shows that the tempotary Japanese translation [いずれかの] (one of those).
Still, thanks for fixing the problem in aux.IsCodeListed.

Lyris12 commented on scripts/DBWS-JP/c100420020.lua:
```diff
@@ -0,0 +1,117 @@
+	e1:SetRange(LOCATION_HAND)
+	e1:SetCountLimit(1,100420020)
+	e1:SetCondition(c100420020.spcon)
+	e1:SetCost(c100420020.spcost)
```
redtext: `Parameter 2 should be "Function".`
```suggestion
```

Lyris12 commented on scripts/DBWS-JP/c100420016.lua:
```diff
@@ -0,0 +1,112 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420016.indestg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420016)==0 end
+	c:RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```
redtext: `attempt to index a nil value (global 'c')`
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420016)==0 end
	e:GetHandler():RegisterFlagEffect(100420016,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/MAZE-EN/c100206006.lua:
```diff
@@ -0,0 +1,75 @@
+	if chkc then return chkc:IsControler(1-tp) and chkc:IsLocation(LOCATION_MZONE) and c100206006.atkfilter(chkc) end
+	if chk==0 then return Duel.IsExistingTarget(c100206006.atkfilter,tp,0,LOCATION_MZONE,1,nil)
+		and e:GetHandler():GetFlagEffect(100206006)==0 end
+	c:RegisterFlagEffect(100206006,RESET_CHAIN,0,1)
```
```suggestion
	e:GetHandler():RegisterFlagEffect(100206006,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420019.lua:
```diff
@@ -0,0 +1,122 @@
+end
+function c100420019.thtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100420019.thfilter,tp,LOCATION_DECK,0,1,nil)
+		and c:GetFlagEffect(100420019)==0 end
+	c:RegisterFlagEffect(100420019,RESET_CHAIN,0,1)
```
```suggestion
		and e:GetHandler():GetFlagEffect(100420019)==0 end
	e:GetHandler():RegisterFlagEffect(100420019,RESET_CHAIN,0,1)
```

Lyris12 commented on scripts/DBWS-JP/c100420021.lua:
```diff
@@ -0,0 +1,117 @@
+	Duel.ShuffleHand(tp)
+end
+function c100420021.imtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100420021)==0 end
+	c:RegisterFlagEffect(100420021,RESET_CHAIN,0,1)
```
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100420021)==0 end
	e:GetHandler():RegisterFlagEffect(100420021,RESET_CHAIN,0,1)
```

Lyris12 reviewed (changes requested):
https://github.com/Fluorohydride/ygopro-pre-script/pull/1061/files#r1134791249
https://github.com/Fluorohydride/ygopro-pre-script/pull/1061/files#r1134790408
https://github.com/Fluorohydride/ygopro-pre-script/pull/1061/files#r1134778647
https://github.com/Fluorohydride/ygopro-pre-script/pull/1061/files#r1134789126
https://github.com/Fluorohydride/ygopro-pre-script/pull/1061/files#r1134786164
https://github.com/Fluorohydride/ygopro-pre-script/pull/1061/files#r1134777412
https://github.com/Fluorohydride/ygopro-pre-script/pull/1061/files#r1134775095

Lyris12 commented on scripts/MAZE-EN/c100206007.lua:
```diff
@@ -0,0 +1,84 @@
+--Labyrinth Wall Shadow
+--scripted by JoyJ
+function c100206007.initial_effect(c)
```
```suggestion
function c100206007.initial_effect(c)
	--Activate
	local e1=Effect.CreateEffect(c)
	e1:SetType(EFFECT_TYPE_ACTIVATE)
	e1:SetCode(EVENT_FREE_CHAIN)
	c:RegisterEffect(e1)
```

Lyris12 reviewed (changes requested):
Bug: "Labyrinth Wall Shadow" cannot be activated from hand

Lyris12 commented on scripts/MAZE-EN/c100206010.lua:
```diff
@@ -0,0 +1,68 @@
+--Prey of the Jirai Gumo
+--scripted by JoyJ
+function c100206010.initial_effect(c)
```
```suggestion
function c100206010.initial_effect(c)
	aux.AddCodeList(c,25955164,62340868,98434877)
```

Lyris12 commented on scripts/MAZE-EN/c100206009.lua:
```diff
@@ -0,0 +1,67 @@
+--Riryoku Guardian
+--scripted by JoyJ
+function c100206009.initial_effect(c)
```
```suggestion
function c100206009.initial_effect(c)
	aux.AddCodeList(c,25955164,62340868,98434877)
```

Lyris12 commented on scripts/MAZE-EN/c100206008.lua:
```diff
@@ -0,0 +1,60 @@
+--Double Attack! Wind and Thunder!!
+--scripted by JoyJ
+function c100206008.initial_effect(c)
```
```suggestion
function c100206008.initial_effect(c)
	aux.AddCodeList(c,25955164,62340868,98434877)
```

Lyris12 reviewed (changes requested):
bug: `Gate Guardian of Thunder and Wind` cannot search "Riryoku Guardian", "Double Attack! Wind and Thunder!!", "Labyrinth Wall Shadow", or "Prey of the Jirai Gumo".

Lyris12 commented on scripts/MAZE-EN/c100206004.lua:
```diff
@@ -0,0 +1,64 @@
+	c:RegisterEffect(e2)
+end
+function c100206004.thfilter(c)
+	return (aux.IsCodeListed(c,25955164) or aux.IsCodeListed(c,62340868)
+		or aux.IsCodeListed(c,98434877)) and c:IsAbleToHand()
```
> This one is weird... I prefer to use "or" because yugioh-wiki shows that the tempotary Japanese translation [いずれかの] (one of those). Still, thanks for fixing the problem in aux.IsCodeListed.
~@IMJoyJ

You're welcome. ... So, there seems to be a conflict in text mechanics, where the OCG text seems to use `or`, whereas the TCG uses `AND`:
> During your Main Phase: You can add 1 Spell/Trap from your Deck to your hand that mentions "Sanga of the Thunder", "Kazejin", **AND** "Suijin".

Lyris12 commented on scripts/MAZE-EN/c100206004.lua:
```diff
@@ -0,0 +1,64 @@
+end
+function c100206004.thfilter(c)
+	return aux.IsCodeListed(c,25955164) and aux.IsCodeListed(c,62340868)
+		and aux.IsCodeListed(c,98434877) and c:IsAbleToHand()
```
> During your Main Phase: You can add 1 **Spell/Trap** from your Deck to your hand that mentions "Sanga of the Thunder", "Kazejin", AND "Suijin".
```suggestion
		and aux.IsCodeListed(c,98434877) and c:IsType(TYPE_SPELL+TYPE_TRAP) and c:IsAbleToHand()
```

Lyris12 reviewed (commented):
Bugs:
> *All Gate Guardian fusion monsters can be summoned by using pieces from your hand. If I understand them correctly, they have to be on the field.
> *Gate Guardian of Wind and Water doesn’t offer negation when an opponent activates a spell card (in my case, Brain Control targeting my Labyrinth Heavy Tank and Swords of Revealing Light) It did work on Mirror Force and Dimensional Prison. Might be just spells that are the problem.
> *Gate Guardian of Water and Thunder effect resolves without offering to target an opponent’s monster or lowering of any monster’s ATK.
> *Labyrinth Wall Shadow blocks monsters from attacking the first turn even if they are level 5 or higher. Also, it allows the destruction of your opponents monster after you select and put a sanga/kazejin/suijin in you spell trap zone and it allows you to target face-down monsters. It should only at the start of your opponent’s battle phase. Not sure if face downs are valid target.

[Source](https://forum.duelistsunite.org/t/gate-guardians-bugs/15950)

IMJoyJ commented:
`All Gate Guardian fusion monsters can be summoned by using pieces from your hand. If I understand them correctly, they have to be on the field.`
Wrongly copied and modified the summon condition from Shadow Ghoul of the Labyrinth. Now fixed.
`Gate Guardian of Wind and Water doesn’t offer negation when an opponent activates a spell card (in my case, Brain Control targeting my Labyrinth Heavy Tank and Swords of Revealing Light) It did work on Mirror Force and Dimensional Prison. Might be just spells that are the problem.`
`Gate Guardian of Water and Thunder effect resolves without offering to target an opponent’s monster or lowering of any monster’s ATK.`
Not sure about these...

`Labyrinth Wall Shadow blocks monsters from attacking the first turn even if they are level 5 or higher. Also, it allows the destruction of your opponents monster after you select and put a sanga/kazejin/suijin in you spell trap zone and it allows you to target face-down monsters. It should only at the start of your opponent’s battle phase. Not sure if face downs are valid target.`
Fixed. It's hard to tell what was I thinking when I write e2 down. Likely drugged.

Lyris12 commented on scripts/DBWS-JP/c100420026.lua:
```diff
@@ -0,0 +1,64 @@
+		and not c:IsAttribute(attr)
+end
+function c100420026.filter(c,e,tp)
+	return c:IsFaceup() c:IsSetCard(0x297) and c:IsAbleToHand()
```
redtext: `'end' expected (to close 'function' at line 19) near 'c'`
```suggestion
	return c:IsFaceup() and c:IsSetCard(0x297) and c:IsAbleToHand()
```

Lyris12 reviewed (changes requested):
> Labrynth Heavy Tank can destroy a monster while there is no “Labyrinth Wall” card on the field.
> Shadow Ghoul of the Labyrinth seems to be unable to use his Graveyard effect when an opponent monster battles.
> Gate Guardians Combined can not special summon Gate Guardian from the Main Deck.
> Gate Guardian of Wind and Water seems to be able to negate only Spell and Trap cards that were on the field before their effect is activated cause I can not negate Infinite Impermanence when activated from the hand and I can negate the effect of Labyrinth Wall Shadow when the opponent try to place a card in their Spell & Trap zone.

IMJoyJ commented:
[Gate Guardians Combined can not special summon Gate Guardian from the Main Deck.]
This is because old card [Gate Guardian]'s setcode is 0x52, not 0x1052.
This can only be fixed by updating cards.cdb.
I think other bugs are fixed now. Thanks for your help.

XGlitchy30 reviewed (changes requested):
Hello, here are some suggestions regarding c.100420036.lua:
- "e1" should have the CATEGORY_GRAVE_ACTION category since it includes an effect that could add a card from the GY to the hand (so Ghost Belle can respond to it)
- Regarding the activation legality, the target function does not check whether 2 cards with different names exist in the player's Deck (it only checks if there are 2 cards that match the filter requirements).
- The "cXXX.filter" function does not specify that the cards must be monsters.
- In the operation function, the "cXXX.thfilter" function is called by passing "attr" (the Attribute of the selected monster) as parameter, which is a property that is irrelevant to the effect's resolution. In addition, you could also use an "if...elseif" block insead of two "if" lines when you define "thfilter".
- The card added by the last part of the effect is not revealed to the opponent (missing Duel.ConfirmCards). Also, the Duel.BreakEffect function should be called after the selection occurs.

Lyris12 commented on scripts/MAZE-EN/c100206003.lua:
```diff
@@ -0,0 +1,72 @@
+	--fusion material
+	c:EnableReviveLimit()
+	aux.AddFusionProcCode3(c,25955164,62340868,98434877,true,true)
+	aux.AddContactFusionProcedure(c,Card.IsAbleToRemoveAsCost,LOCATION_ONFIELD+LOCATION_HAND+LOCATION_REMOVED,0,Duel.Remove,POS_FACEUP,REASON_COST)
```
> Must be Special Summoned (from your Extra Deck) by banishing the above cards from your hand, field, and/or **GY**.
```suggestion
	aux.AddContactFusionProcedure(c,Card.IsAbleToRemoveAsCost,LOCATION_ONFIELD+LOCATION_HAND+LOCATION_GRAVE,0,Duel.Remove,POS_FACEUP,REASON_COST)
```
